### PR TITLE
Memory leak fixes

### PR DESCRIPTION
**Pull request by gbuddappagari**

### CODE REVIEW

File: src/downstream.c [hunk 1]
```diff
@@ -229,6 +229,12 @@ void listenerOnMessage(void * msg, size_t msgSize)
                         free(resp_msg);
                         ParodusPrint("free for downstream decoded msg\n");
                         wrp_free_struct(message);
+                        message = NULL;
+                    }
+                    if(msgType == WRP_MSG_TYPE__REQ && message != NULL)
+                    {
+                        wrp_free_struct(message);
```
Comment (shilpa24balaji) [inline src/downstream.c]: Is this not same as Line 231, as Line 173 include REQ message

Comment (shilpa24balaji) [reply src/downstream.c]: Line 234 is required for REQ success case and not for CRUD

File: src/connection.c [hunk 2]
```diff
@@ -412,9 +412,11 @@ int wait_connection_ready (create_connection_ctx_t *ctx)
   int wait_status;
   char *redirectURL = NULL;
 
-  if(nopoll_conn_wait_for_status_until_connection_ready(get_global_conn(), 10, 
-	&wait_status, &redirectURL)) 
+  if(nopoll_conn_wait_for_status_until_connection_ready(get_global_conn(), 10, &wait_status, &redirectURL))
```
Comment (bill1600) [inline src/connection.c]: resolved in pull req 274

File: src/downstream.c [hunk 3]
```diff
@@ -229,6 +229,12 @@ void listenerOnMessage(void * msg, size_t msgSize)
                         free(resp_msg);
                         ParodusPrint("free for downstream decoded msg\n");
                         wrp_free_struct(message);
+                        message = NULL;
+                    }
+                    if(msgType == WRP_MSG_TYPE__REQ && message != NULL)
```
Comment (bill1600) [inline src/downstream.c]: resolved in pull req 274

File: src/token.c [hunk 4]
```diff
@@ -526,8 +526,13 @@ int allow_insecure_conn(char **server_addr, unsigned int *port)
 		insecure = TOKEN_ERR_ALGO_NOT_ALLOWED;
 	}
 
-	if (insecure >= 0) {
-		ParodusInfo ("JWT claims: %s\n", cJSON_Print (jwt->private_claims));
+	if (insecure >= 0)
+	{
+	    char *temp = NULL;
```
Comment (bill1600) [inline src/token.c]: resolved in pull req 274

**Comment by shilpa24balaji** [comment]
All fixed resolved as part of #274

**Pull request closed without merging**